### PR TITLE
[KOGITO-8692] - :sunflower: Refactor profiles package APIs

### PR DESCRIPTION
**Pull request by ricardozanini**
**Description of the change:**
In this PR, we rearranged the `profiles` package to make it easier to understand and better to accommodate more features and future use cases, such as the addition of new profiles. See #255.

**Motivation for the change:**
See: https://issues.redhat.com/browse/KOGITO-8962

**Checklist**

- [x] Add or Modify a unit test for your change
- [x] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by desmax74 — APPROVED** [review]

**Review by wmedvede — APPROVED** [review]

File: Makefile
Comment (spolti) [inline Makefile:203]: do you want to fail if the target resource already exists?

Comment (ricardozanini) [reply Makefile:203]: It's not failing, but we can change that. The `apply` won't work anymore because the CRD is too long. `kubectl` adds an annotation with everything within the `.spec` if we use `apply`. Hence, the change.

Comment (spolti) [reply Makefile:203]: I see, thanks.

**Review by spolti — APPROVED** [review]

**Review by krisv — APPROVED** [review]

**Comment by ricardozanini** [comment]
Rebased after other merged PRs. Waiting for CR to merge.

**Pull request merged**
